### PR TITLE
fix: allow currentRow for onClick in menuButton item in Table widget

### LINKED ISSUE #26052: [Bug]: Unable to use currentRow within a menu item of a Menu Button of a Table widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have a table with a column which is a menu button. I want to use the "currentRow" variable in that menu button's actions but I'm unable to use it. 

### Steps To Reproduce

1. Drag a table to the canvas. 
2. Add a column with a Menu Button 
3. Add a menu Item to the button, and try to use the `currentRow` property

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

> 1.9.30

### PR DESCRIPTION
**Pull request by SaiCharanChetpelly31**
Fixes #26052 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new property, `customJSControl`, for enhanced configuration in the Table Widget, allowing for dynamic table data handling.
	- Added a new test case to ensure the menu button's `onClick` property correctly accesses the `currentRow` context, improving the robustness of the component.

- **Bug Fixes**
	- Enhanced testing coverage for the table menu button component to ensure correct behavior of dynamic properties.

### CODE REVIEW

**Comment by sagar-qa007** [comment]
@SaiCharanChetpelly31 Can you please explain me for the cypress test case requirement here? Can we test with Unit test ?

**Comment by SaiCharanChetpelly31** [comment]
> @SaiCharanChetpelly31 Can you please explain me for the cypress test case requirement here? Can we test with Unit test ?

@sagar-qa007 For other fields like `visible` and `disabled`, they are checking if `currentRow` is accessible in the Cypress tests itself. I didn't find any unit tests related to this in the Table widget code, which is why I have only written Cypress tests.

You can check the already existing cypress tests for visible and disable fields here
https://github.com/appsmithorg/appsmith/blob/486662a2ed37a9e616d478c5ce296426d522c9c8/app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js#L100C1-L165C8

**Comment by SaiCharanChetpelly31** [comment]
@sagar-qa007 @ApekshaBhosale I hope you're well. I’m curious and excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!
Thank you.
cc: @ramsaptami

**Comment by SaiCharanChetpelly31** [comment]
@sagar-qa007 @ApekshaBhosale I hope you're well. I’m curious and excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!
Thank you.
cc: @ramsaptami

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js [hunk 1]
```diff
@@ -163,5 +163,15 @@ describe(
       });
       cy.get(".table-menu-button-popover li a").should("not.exist");
     });
+    it("5. should check that menuitems onClick property has access to currentRow", () => {
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js:166]: Could you please help me understand the necessity of Cypress test cases in this context? I believe unit testing might be sufficient in this scenario.
Your insights would be appreciated.

Comment (SaiCharanChetpelly31) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js:166]: @sagar-qa007 I have mentioned the reason here.
https://github.com/appsmithorg/appsmith/pull/35381#issuecomment-2268291266

**Comment by SaiCharanChetpelly31** [comment]
@ApekshaBhosale Could you please review this PR?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js (hunk 1 above)
Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/menubutton_spec.js:166]: I agree, this is an integration scenario(Table, property control and eval are involved), we need cypress tests.

**Comment by rahulbarwal** [comment]
@SaiCharanChetpelly31 Please fix merge conflicts.

**Comment by SaiCharanChetpelly31** [comment]
> @SaiCharanChetpelly31 Please fix merge conflicts.

sure, working on it

**Comment by SaiCharanChetpelly31** [comment]
@rahulbarwal I resolved merge conflicts. can you please check?
Thanks in advance.

**Comment by KelvinOm** [comment]
@jsartisan could you please review this PR?

**Comment by jsartisan** [comment]
@rahulbarwal can you check this PR?

**Pull request closed without merging**